cmake/version.cmake

```diff
@@ -0,0 +1,35 @@
+# bgfx versioning scheme:
+# bgfx 1.104.7082
+#      ^ ^^^ ^^^^
+#      | |   +--- Commit number  (https://github.com/bkaradzic/bgfx / git rev-list --count HEAD)
+#      | +------- API version    (from https://github.com/bkaradzic/bgfx/blob/master/scripts/bgfx.idl#L4)
+#      +--------- Major revision (always 1)
+
+# BGFX_API_VERSION generated from https://github.com/bkaradzic/bgfx/blob/master/scripts/bgfx.idl#L4
+# bgfx/src/version.h:
+# BGFX_REV_NUMBER
+# BGFX_REV_SHA1
+
+execute_process(COMMAND git -C bgfx log --pretty=format:'%h' -n 1
```

Comment:
I would suggest using CMake's find module to find the git binary. It's not guaranteed to be installed or in the PATH environment variable. https://cmake.org/cmake/help/v3.0/module/FindGit.html

Reply:
Will implement, ty.

Reply:
Implemented